test/s3select_perf_test.cpp

```diff
@@ -0,0 +1,91 @@
+#include <chrono>
+#include <inttypes.h>
+#include <iostream>
+#include <math.h>
+#include "gperftools/profiler.h"
+#include "s3select_test.h"
+
+#define LOOP_COUNT 100000
+
+void generate_number_csv(std::string& out, size_t size) {
+  std::stringstream ss;
+  for (auto i = 0U; i < size; ++i) {
+    ss << 1 << "," << 2.0 << std::endl;
+    ss << -3 << "," << 4.10 << std::endl;
+    ss << 5 << "," << -6.09 << std::endl;
+    ss << -7 << "," << -8.3 << std::endl;
+    ss << 9 << "," << 10.5 << std::endl;
+    ss << -11 << "," << 12.35 << std::endl;
+    ss << -13 << "," << -14.35 << std::endl;
+    ss << -15 << "," << 16.15 << std::endl;
+  }
+  out = ss.str();
+}
+
+TEST(TestS3SelectPerformance, add)
+{
+  //std::string expected_res = "2007-01-01T00:00:00+00:00";
+  std::string input_csv;
+  size_t size = 100000;
+  generate_number_csv(input_csv, size);
+  std::string input_query = "select add(int(_1), float(_2)) from stdin;" ;
+
+  auto start = std::chrono::high_resolution_clock::now();
+
+  ProfilerStart("./add_perf.txt");
```

Comment:
:+1: 

in that way, it's focused on specific flow(SQL operator)

it's also possible to make it more dynamic, i.e. not to create a TEST per operator.